app/model/SubscriptionOps.scala

```diff
@@ -1,16 +1,50 @@
 package model
 
-import com.gu.memsub.OneOffPeriod
-import com.gu.memsub.subsv2.Subscription
-import com.gu.memsub.subsv2.SubscriptionPlan.PaperPlan
+import com.gu.memsub.{OneOffPeriod, Product, Weekly}
+import com.gu.memsub.subsv2.{PaidCharge, PaidSubscriptionPlan, Subscription}
+import com.gu.memsub.subsv2.SubscriptionPlan.{Paid, PaperPlan, WeeklyPlan}
+import com.github.nscala_time.time.OrderingImplicits._
+import com.typesafe.scalalogging.LazyLogging
+import org.joda.time.LocalDate.now
 
-object SubscriptionOps {
+object SubscriptionOps extends LazyLogging {
 
-  implicit class EnrichedSubscription[P <: PaperPlan](subscription: Subscription[P]) {
-    val latestPlan = {
-      def getLatest(p1: P, p2: P) = if (p1.end.isAfter(p2.end)) p1 else p2
-      subscription.plans.list.reduceLeft(getLatest)
+  type WeeklyPlanOneOff = PaidSubscriptionPlan[Product.Weekly, PaidCharge[Weekly.type, OneOffPeriod]]
+
+  implicit class EnrichedPaidSubscription[P <: Paid](subscription: Subscription[P]) {
+    val currency = subscription.plans.head.charges.currencies.head
+    val nextPlan = subscription.plans.list.maxBy(_.end)
+    val planToManage = {
+      lazy val coveringPlans = subscription.plans.list.filter(p => (p.start.isEqual(now) || p.start.isBefore(now)) && p.end.isAfter(now))
+      lazy val expiredPlans = subscription.plans.list.filter(p => p.end.isBefore(now) || p.end.isEqual(now))
+
+      // The user's sub might be in many situations:
+
+      if (coveringPlans.nonEmpty) {
+        // Has a plan which is currently active (i.e today sits within the plan's start and end dates)
+        coveringPlans.maxBy(_.end)
+      } else if (expiredPlans.nonEmpty) {
+        // Has a plan which has recently expired
+        expiredPlans.maxBy(_.end)
+      } else {
+        // Has a plan(s) starting soon
+        if (subscription.plans.size > 1) {
+          logger.warn("User has multiple future plans, we are showing them their last ending plan")
+        }
+        nextPlan
+      }
     }
-    val renewable = latestPlan.charges.billingPeriod.isInstanceOf[OneOffPeriod]
+  }
+
+  implicit class EnrichedPaperSubscription[P <: PaperPlan](subscription: Subscription[P]) {
+    val renewable = !subscription.autoRenew &&
+      // A sub it not renewable if the customer is waiting to start, or has not yet started their earlier-renewed term
+      subscription.termStartDate.isAfter(now) &&
```

Comment:
the comment is right, but I think the code should say it is renewable if it does *not* start after now